client/ipa/dht.py

```diff
@@ -15,7 +15,9 @@ def __init__(self):
     self._log.info('sensor=%d pin=%d retries=%d' % (Dht._SENSOR, Dht._PIN, Dht._RETRIES))
 
   def get_sample(self):
-    humidity, temperature = Adafruit_DHT.read_retry(Dht._SENSOR, Dht._PIN, retries=Dht._RETRIES)
-    if humidity == None or temperature == None:
-      self._log.warning('failed to read temperature/humidity')
+    humidity = 3300
+    while humidity > 100:

```

Comment:
This is not guaranteed to terminate. Can you try out whether something like 3 retries fixes your issue?

Also we should switch to use read instead of read_retry:
https://github.com/adafruit/Adafruit_Python_DHT/blob/master/Adafruit_DHT/common.py#L65

So if the result is None, or an invalid value outside of [0, 100], wait for 2s and retry, up to Dht._RETRIES.